app/models/subscriber.rb

```diff
@@ -4,12 +4,41 @@ class Subscriber < ApplicationRecord
     validates_uniqueness_of :address, case_sensitive: false
   end
 
+  validate :not_nullified_and_activated
+
   has_many :subscriptions, -> { not_deleted }
   has_many :subscriber_lists, through: :subscriptions
   has_many :digest_run_subscribers, dependent: :destroy
   has_many :digest_runs, through: :digest_run_subscribers
 
+  scope :deactivated, -> { where.not(deactivated_at: nil) }
+  scope :activated, -> { where(deactivated_at: nil) }
+
   def nullify_address!
+    raise "Already nullified." if address.nil?
+    raise "Must be deactivated first." if deactivated_at.nil?
+
     update!(address: nil)
   end
+
+  def deactivate!(datetime: nil)
```

Comment:
Should these be bang methods? They aren't following convention of bang being a dangerous alternative to a non bang version.

I notice they're following the nullify_address! convention but not sure if that itself is an outlier.

Reply:
Yeah, I was following the convention of the old `nullify_address!` method.

I would also argue that since these are calling an `update!` method, it's also following some sort of Rails convention so should be left as bang methods.

Reply:
Sure, it does make sense but that is a tricky convention to follow I think Rails itself just follows the safe / danger one. 

Ideally this'd be something decided at linter level